### PR TITLE
Better error messages on voting failure

### PR DESCRIPTION
**Pull request by jpaddison3**
To alleviate confusion about why users' votes are failing

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203496737758797) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/collections/comments/voting.ts [hunk 1]
```diff
@@ -11,11 +11,17 @@ makeVoteable(Comments, {
   timeDecayScoresCronjob: true,
   customBaseScoreReadAccess,
   userCanVoteOn: (user: DbUser|null, comment: DbComment, voteType: string, extendedVote: any) => {
-    if (!user) return false;
+    if (!user) {
+      return {fail: true, reason: 'You do must be logged in to vote.'};
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/comments/voting.ts:15]: "You must be logged in to vote.", I think?

**Comment by Xodarap** [comment]
I see a bunch of errors like
```
2022-12-14T23:48:06.359Z: [GraphQLError: Boolean cannot represent a non boolean value: { fail: false }] {
  locations: [ { line: 20, column: 3 } ],
  path: [ 'tagRels', 'results', 1, 'currentUserCanVote' ],
  extensions: {
    code: 'INTERNAL_SERVER_ERROR',
    exception: {
      stacktrace: [
        'TypeError: Boolean cannot represent a non boolean value: { fail: false }',
        '    at GraphQLScalarType.serializeBoolean [as serialize] (/mnt/c/Users/bwsit/OneDrive/Documents/GitHub/EAForum2/node_modules/graphql/type/scalars.js:190:9)',
        '    at completeLeafValue (/mnt/c/Users/bwsit/OneDrive/Documents/GitHub/EAForum2/node_modules/graphql/execution/execute.js:635:37)',
        '    at completeValue (/mnt/c/Users/bwsit/OneDrive/Documents/GitHub/EAForum2/node_modules/graphql/execution/execute.js:579:12)',
        '    at /mnt/c/Users/bwsit/OneDrive/Documents/GitHub/EAForum2/node_modules/graphql/execution/execute.js:492:16',   
        '    at processTicksAndRejections (node:internal/process/task_queues:95:5)',
        '    at async Promise.all (index 13)',
        '    at async Promise.all (index 1)',
        '    at async Promise.all (index 0)',
        '    at async Promise.all (index 0)'
      ]
    }
  }
}
```
I can't find any functional impact from these though? This happens when I just look at the homepage

**Review by Xodarap — APPROVED** [review]
astlac

**Comment by jpaddison3** [comment]
> I see a bunch of errors like [...]

🙏

**Pull request merged**